src/appleseed-max-impl/appleseedrenderer/projectbuilder.cpp

```diff
@@ -814,6 +815,43 @@ namespace
             object_assembly_instance->transform_sequence()
                 .set_transform(0.0, transform);
 
+            if (node->GetMotBlurOnOff(time) && node->MotBlur() == 1)
+            {
+                bool is_animated = false;
+                Control* tm_controller;
+                Control* controller;
+                tm_controller = node->GetTMController();
+                auto class_id = tm_controller->ClassID();
+
+                controller = tm_controller->GetPositionController();
+                is_animated = controller->IsAnimated() > 0;
+
+                controller = tm_controller->GetRotationController();
+                is_animated = is_animated || controller->IsAnimated() > 0;
+
+                controller = tm_controller->GetScaleController();
+                is_animated = is_animated || controller->IsAnimated() > 0;
+                
+                if (is_animated)
+                {
+                    float shutter_duration = 0.5f;
+                    if (view_node != nullptr)
```

Comment:
If `view_node` is `nullptr`, `shutter_duration` remains set to 0.5f and so we'll set a transform at time offset `0.5f * GetTicksPerFrame()`. Is that what we want?

Reply:
I think it's actually not quite right, even when there is a Physical Camera in the scene.

It seems to me that we should always set the camera's transforms at t=0 and t=1, and then we should set the camera's shutter times such that:

- Opening times (begin and end) set to `GetShutterOffsetInFrames()` (or 0.0f when there is no Physical Camera)
- Closing times (begin and end) set to `GetShutterOffsetInFrames() + GetShutterDurationInFrames()` (or 0.5f when there is no Physical Camera).

Reply:
Hi @dictoon ,
Are 0 and 1 frame numbers? Can I set transform for 1 to `GetObjTMAfterWSM(time + GetTicksPerFrame())`?
You mentioned camera transforms, but line 837 is about assembly transforms. What do we do here? Can I leave this place as is?
I added changes for camera transforms, only for physical camera case. I think only physical camera has Enable Motion Blur checkbox. Other cameras have it as a viewport multi-pass effect. I don't know if we should consider it or should not.

Reply:
Nope, they're arbitrary times. The idea is as follow: you set transforms (for transformation motion blur) or poses (for deformation motion blur) at arbitrary times, be it seconds, frames, etc. Then you tell the camera when to open the shutter, at what speed (and following what curve) and when to close it. The exact unit for the times is not specified by appleseed, it simply needs to be consistent.